analysis/quantifier-comparisons/01-import-quant-data.Rmd

```diff
@@ -0,0 +1,250 @@
+```
+
+
+### Generate sample info data frame
+
+This generates the sample list from the directory names of the downloads. 
+
+```{r}
+
+quant_info <- c(alevin = alevin_data_dir, 
+                kallisto = kallisto_data_dir,
+                cellranger = cellranger_data_dir) %>%
+  purrr::map(list.dirs, recursive = FALSE, full.names = FALSE) %>%
+  # convert to data frame
+  tibble::enframe(name = "tool", value = "quant_dir") %>%
+  tidyr::unnest(quant_dir) %>%
+  # parse the `quant_dir` into sample and index info
+  tidyr::separate(quant_dir, 
+                  sep = "[-]",
+                  into = c("sample", "index_type"),
+                  remove = FALSE) %>%
+  tidyr::separate(index_type, 
+                  into = c("index_content", "kmer", "decoy"), 
+                  extra = "drop",
+                  fill = "right") %>%
+  # no decoy samples get NA here, convert to text.
+  dplyr::mutate(decoy = tidyr::replace_na(decoy, "no"))
+
+# save info table
+readr::write_tsv(quant_info, quant_info_file )
+# print the info table
+quant_info
+
+```
+
+## Process quantification files
+
+### Alevin
+
+Use tximport to make SCEs
+
+```{r}
+alevin_quant_ids <- quant_info %>%
+  dplyr::filter(tool == "alevin") %>%
+  dplyr::pull(quant_dir)
+
+alevin_sces <- alevin_quant_ids  %>%
+  purrr::map(
+    ~ tximport(file.path(alevin_data_dir, .x, "alevin", "quants_mat.gz"),
+               type = "alevin")) %>%
+  purrr::map(
+    ~ SingleCellExperiment(list(counts = .x$counts))) 
+
+# add names
+names(alevin_sces) <- alevin_quant_ids
+```
+
+### Kallisto
+
+Get file basenames (no extension) for import
+
+```{r}
+kallisto_quant_ids <- quant_info %>%
+  dplyr::filter(tool == "kallisto") %>%
+  dplyr::pull(quant_dir)
+
+base_file <- file.path(kallisto_data_dir, kallisto_quant_ids, "counts", "gene_count")
+```
+
+#### Function for reading Kallisto files
+
+Takes a base file name and reads the the `.mtx` file and associated `.barcodes.txt` and `.genes.txt` files.  
+
+Optionally filters cells/barcodes using `DropletUtils::barcodeRanks()` (set `cellfilter = "knee"`).
+
+This filtering is based on plotting the total UMI count for each barcode against the rank order of those cells by UMI count.
+The point where the UMI count starts to fall rapidly (in log space) relative to the rank change between barcodes is assumed to be a transition point indicating the change from barcodes that capture healthy cells to empty droplets or those only containing only dead cells (see [Lun 2020](https://bioconductor.org/packages/release/bioc/vignettes/DropletUtils/inst/doc/DropletUtils.html#computing-barcode-ranks)).
+By finding the "knee" in this plot and keeping only the barcodes with UMI counts greater than this point, we can enrich for healthy cells.
+Both Alevin and Cell Ranger perform more complicated filtering as part of their default workflows that are based on somewhat more elaborate calculations, but the bulk of the included cells will still above below this threshold.
```

Comment:
I'm not sure what this sentence is supposed to say. I think maybe `still be below this threshold`?  As in, this is more selective than the Alevin and CellRanger retention? (Based on what I remember from what's going in #45...)